Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -1377,6 +1468,7 @@ DataProcessorSpec getAODProducerWorkflowSpec(GID::mask_t src, bool useMC, std::s
   if (src[GID::TPC]) {
     dataRequest->requestClusters(GIndex::getSourcesMask("TPC"), false); // TOF clusters are requested with TOF tracks
   }
+  dataRequest->requestEMCALCells(useMC);
```

Comment:
Looks the request is globally enabled, meaning even if the option to fill EMCAL cells is not enabled the request is performed. I guess this leads to the observation that the fill-calo-cell option has no effect on it. This needs to be handled in a different way, probably as well via `GIndex::getSourcesMask`. @shahor02 Can we add a key for EMCAL in the GIndex?

Reply:
@mfasDa what do you mean? All detectors are there: https://github.com/AliceO2Group/AliceO2/blob/e54e15af180e30c233fe85a278cd077f42bed5e9/DataFormats/Reconstruction/include/ReconstructionDataFormats/GlobalTrackID.h#L47

Reply:
this also looks wrong (depending on flag `useMC`?)

Reply:
@shahor02 I see, then the line simply changes to

``` c++
  if (src[GID::EMC]) {
    dataRequest->requestEMCALCells(useMC);
  }
```

right? + of course the same for filling the calo table ...

Reply:
ok I will implement the changes and remove --fill-calo-cells. @shahor02 you are right that all detectors are there, but i think I also need to add it to GID::mask_t allowedSrc in aod-producer-workflow.cxx

Reply:
@fjonasALICE sure, can you add this in this PR?

Reply:
sure, i will push the changes once my local build finishes